### ISSUE TITLE
Custom `IntEnum` class overloading `__new__` not working correctly

### ISSUE THREAD

**Issue by timrid** [microsoft/pyright#7252]
**Describe the bug**
I overloaded `__new__` of `IntEnum` to add a documentation to each enum value. But this generates a typing error since a few month. I found a workaround (see below) but I'am not sure why my first example is not working.

**Code or Screenshots**
```
import enum
from typing import Self


########################################################
# Example with typing Errors:
########################################################
class IntEnumWithDoc(enum.IntEnum):
    def __new__(cls, val: int, doc: str) -> "Self":
        obj = int.__new__(cls, val)
        obj._value_ = val
        obj.__doc__ = doc
        return obj


class ColorWithTypingErrors(IntEnumWithDoc):
    RED = (1, "This is red.")  # ERROR <- Expression of type "tuple[Literal[1], Literal['This is red.']]" cannot be assigned to declared type "int". "tuple[Literal[1], Literal['This is red.']]" is incompatible with "int"
    BLUE = (2, "This is blue.")  # ERROR <- Expression of type "tuple[Literal[1], Literal['This is red.']]" cannot be assigned to declared type "int". "tuple[Literal[1], Literal['This is red.']]" is incompatible with "int"


########################################################
# Workaround:
########################################################
class EnumWithDoc(enum.Enum):
    def __new__(cls, val: int, doc: str) -> "Self":
        obj = int.__new__(cls, val)
        obj._value_ = val
        obj.__doc__ = doc
        return obj


class ColorWorkaround(int, EnumWithDoc):
    RED = (1010, "This is red.")
    BLUE = (1020, "This is blue.")
```

**VS Code extension or command-line**
VSCode: v1.86.1
pylance: v2024.2.1
Python: 3.11.6

**Comment by erictraut** [comment]
Thanks for the bug report. This is indeed a bug that I introduced recently. Apologies for the regression. It will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
